### PR TITLE
feat(delta-lake): extract table history into operation aspect

### PR DESCRIPTION
**Pull request by MugdhaHardikar-GSLab**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/delta_lake/config.py [hunk 1]
```diff
@@ -76,10 +80,10 @@ def get_complete_path(self):
     def validate_config(cls, values: Dict) -> Dict[str, Any]:
         values["_is_s3"] = is_s3_uri(values["base_path"])
         if values["_is_s3"]:
-            if values["s3"] is None:
+            if values.get("s3") is None:
                 raise ValueError("s3 config must be set for s3 path")
-        values["_complete_path"] = values["base_path"]
-        if values["relative_path"] is not None:
+        values["_complete_path"] = values.get("base_path")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/delta_lake/config.py:85]: looks we have have an access of `values["base_path"]` above as well

File: metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py [hunk 2]
```diff
@@ -122,6 +141,54 @@ def get_fields(self, delta_table: DeltaTable) -> List[SchemaField]:
...
+                    else:
+                        operation_custom_properties[key] = str(val)
+            operation_custom_properties.pop("timestamp", None)
+            operation_custom_properties.pop("operation", None)
+            operation_aspect = OperationClass(
+                timestampMillis=reported_time,
+                lastUpdatedTimestamp=last_updated_timestamp,
+                operationType=statement_type,
+                customOperationType=custom_type
+                if statement_type == OperationTypeClass.CUSTOM
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py:173]: move this logic up to the place where we define `custom_type`

File: metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py [hunk 3]
```diff
@@ -122,6 +141,54 @@ def get_fields(self, delta_table: DeltaTable) -> List[SchemaField]:
 
         return fields
 
+    def _create_operation_aspect_wu(
+        self, delta_table: DeltaTable, dataset_urn: str
+    ) -> Iterable[MetadataWorkUnit]:
+        for hist in delta_table.history(self.source_config.version_history_limit):
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py:147]:
```suggestion
        for hist in delta_table.history(limit=self.source_config.version_history_limit):
```

File: metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py [hunk 4]
```diff
@@ -1,104 +1,12 @@
...
-from datahub.ingestion.source.aws.aws_common import AwsSourceConfig
-from datahub.metadata.schema_classes import GlobalTagsClass, TagAssociationClass
 
 S3_PREFIXES = ["s3://", "s3n://", "s3a://"]
 
 logging.getLogger("py4j").setLevel(logging.ERROR)
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-def get_s3_tags(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py:16]: why are all of these changes here?

**Review by hsheth2 — DISMISSED** [review]
Left some minor comments

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**